php-frontend/src/test/java/org/sonar/plugins/php/api/tests/PHPCheckVerifierTest.java

```diff
@@ -0,0 +1,163 @@
+        @Override
+        public void visitCompilationUnit(CompilationUnitTree tree) {
+          context().newIssue(this, "File issue").cost(2.5d);
+        }
+      });
+  }
+
+  @Test(expected = ComparisonFailure.class)
+  public void legacy_file_issue_wrong_message() throws Exception {
+    PHPCheckVerifier.verify(new File("src/test/resources/tests/file-issue.php"),
+      new PHPVisitorCheck() {
+        @Override
+        public void visitCompilationUnit(CompilationUnitTree tree) {
+          context().newIssue(this, "Wrong message").cost(2.5d);
+        }
+      });
+  }
+
+  @Test(expected = ComparisonFailure.class)
+  public void legacy_file_issue_wrong_cost() throws Exception {
+    PHPCheckVerifier.verify(new File("src/test/resources/tests/file-issue.php"),
+      new PHPVisitorCheck() {
+        @Override
+        public void visitCompilationUnit(CompilationUnitTree tree) {
+          context().newIssue(this, "File issue").cost(5.2d);
+        }
+      });
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void legacy_file_issue_wrong_check() throws Exception {
+    PHPCheckVerifier.verify(new File("src/test/resources/tests/file-issue.php"),
+      new PHPVisitorCheck() {
+        @Override
+        public void visitCompilationUnit(CompilationUnitTree tree) {
+          PHPVisitorCheck otherCheck = new PHPVisitorCheck() {
+          };
+          context().newIssue(otherCheck, "File issue").cost(2.5d);
+        }
+      });
+  }
+
+  @Test
+  public void legacy_line_issue() throws Exception {
+    PHPCheckVerifier.verify(new File("src/test/resources/tests/line-issue.php"),
+      new PHPVisitorCheck() {
+        @Override
+        public void visitCompilationUnit(CompilationUnitTree tree) {
+          context().newIssue(this, "Line issue").line(2);
+        }
+      });
+  }
+
+  @Test(expected = ComparisonFailure.class)
+  public void legacy_line_issue_wrong_message() throws Exception {
+    PHPCheckVerifier.verify(new File("src/test/resources/tests/line-issue.php"),
+      new PHPVisitorCheck() {
+        @Override
+        public void visitCompilationUnit(CompilationUnitTree tree) {
+          context().newIssue(this, "Wrong message").line(2);
+        }
+      });
+  }
+
+  @Test
+  public void file_issue() throws Exception {
+    PHPCheckVerifier.verify(new File("src/test/resources/tests/file-issue.php"),
+      new PHPVisitorCheck() {
+        @Override
+        public void visitCompilationUnit(CompilationUnitTree tree) {
+          context().newFileIssue(this, "File issue").cost(2.5d);
+        }
+      });
+  }
+
+  @Test
+  public void line_issue() throws Exception {
+    PHPCheckVerifier.verify(new File("src/test/resources/tests/line-issue.php"),
+      new PHPVisitorCheck() {
+        @Override
+        public void visitCompilationUnit(CompilationUnitTree tree) {
+          context().newLineIssue(this, 2, "Line issue");
+        }
+      });
+  }
+
+  @Test
+  public void precise_issue() throws Exception {
```

Comment:
Why not a test to verify that we can have a ComparisonFailure on a precise issue?